src/hnsw/build.c

```diff
@@ -79,7 +79,15 @@ static void AddTupleToUsearchIndex(ItemPointer tid, Datum *values, HnswBuildStat
     if(buildstate->hnsw != NULL) hnsw_add(buildstate->hnsw, vector, label);
 #endif
 #ifdef LANTERN_USE_USEARCH
-    if(buildstate->usearch_index != NULL) usearch_add(buildstate->usearch_index, label, vector, usearch_scalar, &error);
+    if(buildstate->usearch_index != NULL) {
+        size_t capacity;
+        capacity = usearch_capacity(buildstate->usearch_index, &error);
+        if(capacity == usearch_size(buildstate->usearch_index, &error)) {
+            usearch_reserve(buildstate->usearch_index, 2 * capacity, &error);
```

Comment:
Will reallocating a memory of (capacity + estimated size of one page) give some benefits here?
@Ngalstyan4 @ezra-varady